save-cosv/src/main/kotlin/com/saveourtool/save/cosv/storage/RawCosvFileStorage.kt

```diff
@@ -63,13 +68,22 @@ class RawCosvFileStorage(
     /**
      * @param organizationName
      * @param userName
-     * @return count of all [RawCosvFileDto]s which belongs to [organizationName] and uploaded by [userName]
+     * @return statistics [RawCosvFileStatisticDto] for all [RawCosvFileDto]s which belongs to [organizationName] and uploaded by [userName]
      */
-    fun countByOrganizationAndUser(
+    fun statisticsByOrganizationAndUser(
         organizationName: String,
         userName: String,
-    ): Mono<Long> = blockingToMono {
-        s3KeyManager.countByOrganizationAndUser(organizationName, userName)
+    ): Mono<RawCosvFileStatisticsDto> = blockingToMono {
+        val filesList = s3KeyManager.listByOrganizationAndUser(organizationName, userName).toList()
+        RawCosvFileStatisticsDto(
+            s3KeyManager.countByOrganizationAndUser(organizationName, userName),
+            filesList.count { it.isUploadedZipArchive() }.toLong(),
+            filesList.count { it.isUploadedJsonFile() }.toLong(),
+            s3KeyManager.countByOrganizationAndUserAndStatus(organizationName, userName, RawCosvFileStatus.PROCESSED) +
```

Comment:
you already fetched all `filesList` -- maybe use them instead of additional calls to database

Reply:
Done